tests/integration/cmd_dev_test.go

```diff
@@ -2658,47 +2659,28 @@ CMD ["npm", "start"]
 		}))
 	}
 
-	Context("using Kubernetes cluster", func() {
+	Context("using a default namespace", func() {
 		BeforeEach(func() {
-			if os.Getenv("KUBERNETES") != "true" {
-				Skip("This is a Kubernetes specific scenario, skipping")
-			}
-		})
-
-		It("should run odo dev successfully on default namespace", func() {
-			helper.CopyExample(filepath.Join("source", "nodejs"), commonVar.Context)
-			helper.Cmd("odo", "init", "--name", cmpName, "--devfile-path", helper.GetExamplePath("source", "devfiles", "nodejs", "devfile.yaml")).ShouldPass()
-			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
-
-			session, _, errContents, _, err := helper.StartDevMode(helper.DevSessionOpts{})
-			Expect(err).ToNot(HaveOccurred())
-			defer func() {
-				session.Stop()
-				session.WaitEnd()
-			}()
-			helper.DontMatchAllInOutput(string(errContents), []string{"odo may not work as expected in the default project"})
+			commonVar.CliRunner.SetProject("default")
 		})
-	})
-
-	/* TODO(feloy) Issue #5591
-	Context("using OpenShift cluster", func() {
-		BeforeEach(func() {
-			if os.Getenv("KUBERNETES") == "true" {
-				Skip("This is a OpenShift specific scenario, skipping")
-			}
+		AfterEach(func() {
+			commonVar.CliRunner.SetProject(commonVar.Project)
 		})
-		It("should run odo dev successfully on default namespace", func() {
-			helper.CopyExample(filepath.Join("source", "nodejs"), commonVar.Context)
-			helper.Cmd("odo", "init", "--name", cmpName, "--devfile-path", helper.GetExamplePath("source", "devfiles", "nodejs", "devfile.yaml")).ShouldPass()
-			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
-
-			session, _, errContents, err := helper.StartDevMode(helper.DevSessionOpts{})
-			Expect(err).ToNot(HaveOccurred())
-			defer session.Stop()
-			helper.MatchAllInOutput(string(errContents), []string{"odo may not work as expected in the default project"})
+		When("a component is created", func() {
+			BeforeEach(func() {
+				helper.CopyExample(filepath.Join("source", "nodejs"), commonVar.Context)
+				helper.Cmd("odo", "init", "--name", cmpName, "--devfile-path", helper.GetExamplePath("source", "devfiles", "nodejs", "devfile.yaml")).ShouldPass()
+				helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
+			})
+			It("should print warning about default namespace when running odo dev", func() {
+				namespace := "project"
+				if helper.IsKubernetesCluster() {
+					namespace = "namespace"
+				}
+				helper.WaitForDevModeToContain(helper.DevSessionOpts{}, fmt.Sprintf("You are using \"default\" %[1]s, odo may not work as expected in the default %[1]s.", namespace), true)
```

Comment:
Since the Dev Session would be started successfully, I am wondering if using `helper.RunDevMode` or `helper.StartDevMode`  wouldn't be enough for checking the standard error output. For example:

```suggestion
				err := helper.RunDevMode(helper.DevSessionOpts{}, func(session *gexec.Session, outContents []byte, errContents []byte, ports map[string]string) {
					Expect(string(errContents)).Should(ContainSubstring("You are using \"default\" %[1]s, odo may not work as expected in the default %[1]s.", namespace))
				})
				Expect(err).ShouldNot(HaveOccurred())
```